src/composed/signed_key/parse.rs

```diff
@@ -54,18 +54,38 @@ pub fn from_armor_many<'a, R: io::Read + io::Seek + 'a>(
 pub fn from_bytes_many<'a>(
     bytes: impl io::Read + 'a,
 ) -> Box<dyn Iterator<Item = Result<PublicOrSecret>> + 'a> {
+    // Once we encounter a packet that we consider a hard error, this is set to true.
+    // If true, we don't process any subsequent packets.
+    let mut hard_error = false;
+
     let packets = PacketParser::new(bytes)
-        .filter_map(|p| {
+        .filter_map(move |p| {
+            if hard_error {
+                return None;
+            };
+
             match p {
                 Ok(Packet::Marker(_m)) => {
                     debug!("skipping marker packet");
                     None
                 }
-                Ok(p) => Some(p),
-                Err(_) => {
-                    // for now we are skipping any packets that we failed to parse
-                    warn!("skipping packet: {:?}", p);
-                    None
+                Ok(p) => Some(Ok(p)),
+                Err(e) => {
+                    if let Error::InvalidPacketContent(b) = &e {
```

Comment:
maybe this could be a helper function, given it is used in multiple places already, and we will want to keep this logic consistent likely